Projects/Server/Network/NetState/NetState.cs

```diff
@@ -533,9 +543,36 @@ private void LogPacket(ReadOnlySpan<byte> buffer, bool incoming)
         }
     }
 
+    /**
+     * To be called when a packet handler wishes to continue execution
+     * asynchronously.  As long as this #Task runs, no more packets
+     * are handled.
+     */
+    public void AwaitPacketHandlerTask(Task task, CancellationTokenSource cancel)
```

Comment:
What is wrong with async/await and a try/catch? Since the code starts on the game thread, which has a synchronization context that pushes continuations back to the game thread, I don't think this is necessary?